source/grid/manifold_lib.cc

```diff
@@ -23,6 +23,114 @@
 
 DEAL_II_NAMESPACE_OPEN
 
+
+
+
+
+template <int dim,
+          int dim_A, int spacedim_A, int chartdim_A,
+          int dim_B, int spacedim_B, int chartdim_B>
+TensorProductManifold<dim, dim_A, spacedim_A, chartdim_A, dim_B, spacedim_B, chartdim_B>
+::TensorProductManifold(
+  const ChartManifold<dim_A, spacedim_A, chartdim_A> &manifold_A,
+  const ChartManifold<dim_B, spacedim_B, chartdim_B> &manifold_B)
+  : manifold_A (manifold_A), manifold_B (manifold_B)
+{}
```

Comment:
Here you neglect altogether periodicity of the resulting manifold. This should be a Point<chartdim_A+chartdim_B> that combines the periodicities of `manifold_A` and `manifold_B`.